codecs/jxl/enc/jxl_enc.cpp

```diff
@@ -0,0 +1,57 @@
+#include <emscripten/bind.h>
+#include <emscripten/val.h>
+
+#include "jxl/enc_file.h"
+
+using namespace emscripten;
+
+thread_local const val Uint8Array = val::global("Uint8Array");
+
+struct JXLOptions {
+  // 1 = slowest
+  // 7 = fastest
+  int speed;
+};
+
+val encode(std::string image, int width, int height, JXLOptions options) {
+  jxl::CompressParams cparams;
+  jxl::PassesEncoderState passes_enc_state;
+  jxl::CodecInOut io;
+  jxl::PaddedBytes bytes;
+  jxl::ImageBundle* main = &io.Main();
+
+  cparams.speed_tier = static_cast<jxl::SpeedTier>(options.speed);
```

Comment:
Maybe do `10-options.speed` so it behaves the same as in `cjpegxl`(and set the default somewhere in the middle instead of very-slow or very-fast).